modules/machineset-azure-confidential-vms.adoc

```diff
@@ -0,0 +1,91 @@
+// * machine_management/control_plane_machine_management/cpmso-using.adoc
+
+ifeval::["{context}" == "cpmso-using"]
+:cpmso:
+endif::[]
+
+:_content-type: PROCEDURE
+[id="machineset-azure-confidential-vms_{context}"]
+= Configuring Azure confidential virtual machines by using machine sets
+
+{product-title} {product-version} supports Azure confidential virtual machines (VMs).
+
+[NOTE]
+====
+Confidential VMs are currently not supported on 64-bit ARM architectures.
+====
+
+By editing the machine set YAML file, you can configure the confidential VMs options that a machine set uses for machines that it deploys. These machines can use UEFI security features such as Secure Boot or a dedicated virtual Trusted Platform Module (vTPM) instance.
+
+ifdef::cpmso[]
+[IMPORTANT]
+====
+Not all instance types support confidential VMs. Do not change the instance type for a control plane machine set that is configured to use confidential VMs to a type that is incompatible. Using an incompatible instance type can cause your cluster to become unstable.
+====
+endif::cpmso[]
+
+For more information about related features and functionality, see the Microsoft Azure documentation about link:https://learn.microsoft.com/en-us/azure/confidential-computing/confidential-vm-overview[Confidential virtual machines].
+
+.Procedure
+
+. In a text editor, open the YAML file for an existing machine set or create a new one.
+
+. Edit the following section under the `providerSpec` field:
++
+--
+.Sample configuration
+[source,yaml]
+----
+ifndef::cpmso[]
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+endif::cpmso[]
+ifdef::cpmso[]
+apiVersion: machine.openshift.io/v1
+kind: ControlPlaneMachineSet
+endif::cpmso[]
+# ...
+spec:
+  template:
+    spec:
+      providerSpec:
+        value:
+          osDisk:
+            # ...
+            managedDisk:
+              securityProfile: # <1>
+                securityEncryptionType: VMGuestStateOnly # <2>
+            # ...
+          securityProfile: # <3>
+            settings:
+                securityType: ConfidentialVM # <4>
+                confidentialVM:
+                  uefiSettings: # <5>
+                    secureBoot: Disabled # <6>
+                    virtualizedTrustedPlatformModule: Enabled # <7>
+          vmSize: Standard_DC16ads_v5 # <8>
+# ...
+----
+<1> Specifies security profile settings for the managed disk when using a confidential VM.
+<2> Specifies the type of encryption used for a confidential VM. The following values are valid:
++
+* `DiskWithVMGuestState` enables encryption of the manged disk and the Azure VM Guest State (VMGS) blob with a platform-managed key (PMK) or a customer-managed key (CMK). This setting requires the use of vTPM and secure boot.
+* `VMGuestStateOnly` enables encryption of the VMGS blob. This setting requires the use of vTPM.
+<3> Specifies security profile settings for the managed disk when using a confidential VM.
```

Comment:
Should `<3>` also be "for the managed disk" (like `<1>`)?

Reply:
`<3>` refers to the `securityProfile` of the Virtual Machine, while `<1>` refers to the `securityProfile` of the managed disk.

> I understand that it could be confusing, but this is how the Azure API is defined and we decided not to diverge too much from it.

Reply:
Thanks :) confusion is part of why I asked - I think if we label them clearly it will be as good as it can be. Will update with that.